projects/dsp-ui/src/lib/viewer/representation/upload-form/upload-form.component.ts

```diff
@@ -0,0 +1,134 @@
+import { Component, Input, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { NotificationService } from '../../../action/services/notification.service';
+import { UploadedFileResponse, UploadFileService } from '../../services/upload-file.service';
+
+@Component({
+    selector: 'dsp-upload-form',
+    templateUrl: './upload-form.component.html',
+    styleUrls: ['./upload-form.component.scss']
+})
+export class UploadFormComponent implements OnInit {
+
+    @Input() readonly resourceTyoe = 'Image'; // only StillImageRepresentation supported so far
+    @Output() file: File;
+
+    readonly fromLabels = {
+        upload: 'Upload file',
+        drag_drop: 'Drag and drop or click to upload'
+    };
+    form: FormGroup;
+    get fileControl() { return this.form.get('file') as FormControl; }
+    isLoading = false;
+    thumbnaillUrl: string;
+
+    constructor(
+        private readonly _fb: FormBuilder,
+        private readonly _ns: NotificationService,
+        private readonly _ufs: UploadFileService
+    ) { }
+
+    ngOnInit(): void {
+        this.initializeForm();
+    }
+
+    initializeForm(): void {
+        this.form = this._fb.group({
+            file: [undefined, Validators.required]
+        }, { updateOn: 'blur' });
+    }
+
+    addFile(event): void {
+        let files: File[] = [];
+        files = event.target?.files ? event.target.files : event;
+        if (this.isMoreThanOneFile(files)) {
+            const error = 'ERROR: Only one file allowed at a time';
+            this._ns.openSnackBar(error);
+            this.file = null;
+        } else {
+            const formData = new FormData();
+            this.file = files[0];
+            if (!this.isFileTypeSupported(this.file.type)) {
+                const error = 'ERROR: File type not supported';
+                this._ns.openSnackBar(error);
+                this.file = null;
+            } else {
+                // show loading indicator only for files > 1MB
+                this.isLoading = this.file.size > 1048576 ? true : false;
+                formData.append(this.file.name, this.file);
+                this._ufs.upload(formData).subscribe(
+                    (res: UploadedFileResponse) => {
+                        const tempUrl = res.uploadedFiles[0].temporaryUrl;
+                        const thumbUri = `${tempUrl}/full/150,/0/default.jpg`;
+                        this.thumbnaillUrl = `${thumbUri}`.replace('http://sipi:1024/', this._ufs.envUrl);
```

Comment:
I think in production, "http://sipi:1024/" would be different.

Reply:
This is actually good question if it depends on the env? I mean according to the docs it might, since there is **sipihost** mentioned. But base on what I got as response: `http://sipi:1024/` I assume it might be const. I will ask how does it work.

Reply:
Fixed with [this PR](https://github.com/dasch-swiss/knora-api/pull/1751) and d1699c8

Reply:
Thanks. So if I understand correctly, the is no hard coded expectation anymore. Instead, it comes from the config, right?

Reply:
Correct.